WW-200 Community Page discussions module

dianafa opened the pull request.

Commit:
XW | fix counter

hakubo committed:
Merge pull request #11371 from Wikia/XW-2068-top-community

XW-2068 | related wiki module

hakubo committed:
limit width of an image on cross wiki search page

Commit:
XW-search | rename exactResult to exactWikiMatch

Commit:
XW | add tracking for exact wiki match

hakubo committed:
change exact-match to exact-wiki-match

Commit:
XW-search | download some new messages

Commit:
XW-search | download a fixed message

Commit:
Merge branch 'dev' into SOC-2815

Commit:
SOC-2815 update formatting

Commit:
SOC-2815 remove commented out code

mleczey committed:
SOC-2866 | fix: add missing border around avatar image

mleczey committed:
SOC-2898 | refactor: changes after code review

mleczey committed:
SOC-2864 | fix: chagne images to biggers ones in order to improve their look on retina

mleczey committed:
SOC-2864 | fix: switch top bar background solution to image

mleczey committed:
SOC-2864 | fix: change top bar height to 5px

mleczey committed:
SOC-2864 | fix: change footer and header images dimensions

mleczey committed:
SOC-2866 | fix: contribution appreciation email text alignment change to left

mleczey committed:
SOC-2866 | fix: remove typo

mleczey committed:
SOC-2864 | fix: remove padding on gmail client for top bar

Commit:
WW-200 Display discussions widget on community page

Commit:
Remove unneeded param

Commit:
Remove logs, remove unnecessary commas

mleczey committed:
SOC-2864 | fix: remove padding on gmail client for top bar

mleczey committed:
SOC-2864 | fix: add height for top bar in markup

mleczey committed:
SOC-2864 | fix: make top bar full width on outlook

rybmat committed:
cr

Commit:
WIP: make the discussions widget styled according to the designs

rybmat committed:
moving var declaration

Commit:
ADEN-3594 Change playwire sizes and adunit

mleczey committed:
SOC-3086 | feat: underline links in footer

drets committed:
ADEN-3594 Introduce ad unit builder

Commit:
Styling fixes to show only three lines of post text

Commit:
Design review changes
Code review changes

mleczey committed:
SOC-3059 | refactor: change social links for different languages

mleczey committed:
SOC-3059 | feat: fix for instagram icon link

drets committed:
ADEN-3594 Use iframe for getting player size

Commit:
ADEN-3594 Get aspect ratio from creative parameters

drets committed:
ADEN-3594 Simplify test

Commit:
Design Review changes

Commit:
FAN-1027 Use different links in global nav for user on Fandom stories

rybmat committed:
labels renaming

lizlux committed:
FAN-1027 fixing link order

mleczey committed:
SOC-3059 | refactor: reverrting and remvoing all social links for languages different than en

mleczey committed:
SOC-3059 | style: remove empty lines

drets committed:
ADEN-3594 CR fixes

Commit:
XW-2101 | Search - setValues

Commit:
spaces to tabs

Commit:
XW-2101 | Search fandom- fix XSS voulnerabilities

Commit:
XW-2101 | Search - address CR in FandomSearch

Commit:
XW-2101 | ResultHelper.php fix whitespaces

Commit:
XW-2101 | Search topArticles - fix XSSes

Commit:
XW-2101 | Search address CR

Commit:
XW-2101 | Search fix issues in Tracking.js

Commit:
Merge remote-tracking branch 'origin/dev' into XW-search

hakubo committed:
remove code for old GlobalFooter

hakubo committed:
remove CorporateFooter

drets committed:
ADEN-3594 Alphabetical order

pgrabowski-wikia committed:
adding confirmation in case a searched element does not exist

Commit:
XW-2107 | Close search input on Escape key press

Commit:
XW-2101 | Search - tests for Fandom API parsing

Commit:
ADEN-3594 Add default values for build function

Commit:
XW-2107 | use `which` instead of `keyCode`

Commit:
XW-2101 | Search fix unit tests

Commit:
XW-2107 | more responsiveness: use `keydown` instead of `keyup`

Commit:
XW-2107 | consistence with Mercury: use `keyup` instead of `keydown` (reverts 404c0b4)

Commit:
ADEN-3994 calc the page height

Commit:
XW-2101 | Move shortenNumberDecorator to Language

dianafa committed:
WW-200 remove static function calls

dianafa committed:
fix php docs and adjust to coding convntions

Commit:
Merge branch 'dev' into FAN-1027

dianafa committed:
WW-200 make render accept also int values

Commit:
FAN-1027 Show verticals on Fandom product for all languages

Commit:
SUS-1048: cap date range instead of throwing exception

dianafa committed:
improve readabilty of code, change params order

WikiaTech commented on extensions/wikia/EmbeddableDiscussions/scripts/templates.mustache.js:
```diff
@@ -1,4 +1,4 @@
 /* jshint ignore:start */ define( 'embeddablediscussions.templates.mustache', [], function() { 'use strict'; return {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Reformat the code to have only one statement per line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AOneStatementPerLine)

WikiaTech commented on extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php:
```diff
@@ -36,134 +33,170 @@ public static function onBeforePageDisplay() {
 		) {
-			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', 'mostrecent' )->plain() .
-				wfMessage( 'embeddable-discussions-parameter-error-boolean' )->plain();
+			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error',
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "embeddable-discussions-parameter-error" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech commented on extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php:
```diff
@@ -36,134 +33,170 @@ public static function onBeforePageDisplay() {
+	 * @return string
+	 */
+	private function renderError( $errorMessage ) {
+		$templateEngine = ( new Wikia\Template\MustacheEngine )->setPrefix( __DIR__ . '/templates' );
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "/templates" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech commented on extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php:
```diff
@@ -36,134 +33,170 @@ public static function onBeforePageDisplay() {
 	 * @return true if ok, false if error
 	 */
-	private function checkArguments( array $args, $modelData, &$errorMessage ) {
+	private function checkArguments( $modelData, array $args, &$errorMessage ) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)

WikiaTech commented on extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php:
```diff
@@ -36,134 +33,170 @@ public static function onBeforePageDisplay() {
 		// columns must be integer in range
-		if ( isset( $args['columns'] ) ) {
-			$columns = $args['columns'];
+		if ( isset( $args[ 'columns' ] ) ) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "columns" 6 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech commented on extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php:
```diff
@@ -36,134 +33,170 @@ public static function onBeforePageDisplay() {
-		if ( isset( $args['mostrecent'] ) &&
-			$args['mostrecent'] !== 'true' &&
-			$args['mostrecent'] !== 'false'
+		if ( isset( $args[ 'mostrecent' ] ) &&
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "mostrecent" 6 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech commented:
SonarQube analysis reported 15 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 4 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 11 minor

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [CommunityPageSpecialController.class.php#L3](https://github.com/Wikia/app/blob/a66aaa4e39950a5b8d485465285b450ebbeb7c67/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L3): Class "CommunityPageSpecialController" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [ext.embeddableDiscussions.js#L10](https://github.com/Wikia/app/blob/a66aaa4e39950a5b8d485465285b450ebbeb7c67/extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js#L10): Function has 8 parameters which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [ext.embeddableDiscussions.js#L69](https://github.com/Wikia/app/blob/a66aaa4e39950a5b8d485465285b450ebbeb7c67/extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js#L69): Restrict what this loop acts on by testing each property. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AForIn)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L38](https://github.com/Wikia/app/blob/a66aaa4e39950a5b8d485465285b450ebbeb7c67/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L38): Define a constant instead of duplicating this literal "modules" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L47](https://github.com/Wikia/app/blob/a66aaa4e39950a5b8d485465285b450ebbeb7c67/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L47): Define a constant instead of duplicating this literal "CommunityPageSpecialController" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L153](https://github.com/Wikia/app/blob/a66aaa4e39950a5b8d485465285b450ebbeb7c67/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L153): Define a constant instead of duplicating this literal "noAdminContactText" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L153](https://github.com/Wikia/app/blob/a66aaa4e39950a5b8d485465285b450ebbeb7c67/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L153): Define a constant instead of duplicating this literal "communitypage-no-admins-contact" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L353](https://github.com/Wikia/app/blob/a66aaa4e39950a5b8d485465285b450ebbeb7c67/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L353): Define a constant instead of duplicating this literal "timeAgo" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech commented on extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php:
```diff
@@ -36,134 +33,170 @@ public static function onBeforePageDisplay() {
-			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', $args['category'] )->plain() .
-				wfMessage( 'embeddable-discussions-parameter-error-category' )->plain();
+		if ( !empty( $modelData[ 'invalidCategory' ] ) ) {
+			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', $args[ 'category' ],
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "category" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

lukaszkonieczny commented on extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php:
```diff
@@ -36,134 +33,170 @@ public static function onBeforePageDisplay() {
-			$args['mostrecent'] !== 'false'
+		if ( isset( $args[ 'mostrecent' ] ) &&
+			$args[ 'mostrecent' ] !== 'true' &&
+			$args[ 'mostrecent' ] !== 'false'
```
Can we check `is_bool( $args[ 'mostrecent' ] )` ?

lukaszkonieczny commented on extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php:
```diff
@@ -36,134 +33,170 @@ public static function onBeforePageDisplay() {
+
+		$showLatest = !empty( $args[ 'mostrecent' ] ) && filter_var( $args[ 'mostrecent' ], FILTER_VALIDATE_BOOLEAN );
+		$itemCount = !empty( $args[ 'size' ] ) ? intval( $args[ 'size' ] ) : self::ITEMS_DEFAULT;
+		$columns = !empty( $args[ 'columns' ] ) ? intval( $args[ 'columns' ] ) : self::COLUMNS_DEFAULT;
```
Aren't these fields also checked in `checkArguments` method?

dianafa committed:
WW-200 move params to consts

lukaszkonieczny commented on extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php:
```diff
@@ -9,17 +9,14 @@ class EmbeddableDiscussionsController {
-	private $templateEngine;
-
-	function __construct() {
-		$this->templateEngine = ( new Wikia\Template\MustacheEngine )->setPrefix( __DIR__ . '/templates' );
```
Why did you remove it and added in 3 different places?

WikiaTech commented on extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php:
```diff
@@ -36,134 +38,171 @@ public static function onBeforePageDisplay() {
 		) {
-			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', 'mostrecent' )->plain() .
-				wfMessage( 'embeddable-discussions-parameter-error-boolean' )->plain();
+			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error',
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "embeddable-discussions-parameter-error" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech commented on extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php:
```diff
@@ -36,134 +38,171 @@ public static function onBeforePageDisplay() {
-		if ( isset( $args['mostrecent'] ) &&
-			$args['mostrecent'] !== 'true' &&
-			$args['mostrecent'] !== 'false'
+	private function checkArguments( $modelData, array $args, &$errorMessage ) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)

WikiaTech commented on extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php:
```diff
@@ -9,17 +9,19 @@ class EmbeddableDiscussionsController {
+	const PARAM_MOSTRECENT = 'mostrecent';
+	const PARAM_SIZE = 'size';
+	const PARAM_COLUMNS = 'columns';
+	const PARAM_CATEGORY = 'category';
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "category" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech commented:
SonarQube analysis reported 13 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 4 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 9 minor

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [CommunityPageSpecialController.class.php#L3](https://github.com/Wikia/app/blob/172f46a25875a20bc6e2a9f0e7fb3aae5e4493c4/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L3): Class "CommunityPageSpecialController" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [ext.embeddableDiscussions.js#L10](https://github.com/Wikia/app/blob/172f46a25875a20bc6e2a9f0e7fb3aae5e4493c4/extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js#L10): Function has 8 parameters which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [ext.embeddableDiscussions.js#L69](https://github.com/Wikia/app/blob/172f46a25875a20bc6e2a9f0e7fb3aae5e4493c4/extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js#L69): Restrict what this loop acts on by testing each property. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AForIn)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L38](https://github.com/Wikia/app/blob/172f46a25875a20bc6e2a9f0e7fb3aae5e4493c4/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L38): Define a constant instead of duplicating this literal "modules" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L47](https://github.com/Wikia/app/blob/172f46a25875a20bc6e2a9f0e7fb3aae5e4493c4/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L47): Define a constant instead of duplicating this literal "CommunityPageSpecialController" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L153](https://github.com/Wikia/app/blob/172f46a25875a20bc6e2a9f0e7fb3aae5e4493c4/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L153): Define a constant instead of duplicating this literal "noAdminContactText" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L153](https://github.com/Wikia/app/blob/172f46a25875a20bc6e2a9f0e7fb3aae5e4493c4/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L153): Define a constant instead of duplicating this literal "communitypage-no-admins-contact" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L353](https://github.com/Wikia/app/blob/172f46a25875a20bc6e2a9f0e7fb3aae5e4493c4/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L353): Define a constant instead of duplicating this literal "timeAgo" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

lukaszkonieczny commented on extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php:
```diff
@@ -36,134 +38,171 @@ public static function onBeforePageDisplay() {
+			'upvoteRequestUrl' => $modelData[ 'upvoteRequestUrl' ],
+		] );
+
+		$modelData[ 'columnsWrapperClass' ] = $columns === 2 ? 'embeddable-discussions-threads-columns' : '';
```
Can we replace `2` with some const?

lukaszkonieczny commented on extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js:
```diff
@@ -147,10 +147,12 @@ require([
 		});
 
-		$('.embeddable-discussions-module').on('click', '.upvote', function(event) {
+		var discussionsModule = $('.embeddable-discussions-module');
```
variables should be declared at the beginning of the function

lukaszkonieczny commented on extensions/wikia/EmbeddableDiscussions/templates/DiscussionThreads.mustache:
```diff
@@ -22,9 +19,11 @@
 			<div class="embeddable-discussions-title">
 				{{title}}
 			</div>
+            {{/title}}
```
Please use tabs instead of spaces

rybmat committed:
label naming

dianafa committed:
make templateEngine private var

WikiaTech commented on extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php:
```diff
@@ -9,6 +9,11 @@ class EmbeddableDiscussionsController {
+	const PARAM_MOSTRECENT = 'mostrecent';
+	const PARAM_SIZE = 'size';
+	const PARAM_COLUMNS = 'columns';
+	const PARAM_CATEGORY = 'category';
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "category" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech commented:
SonarQube analysis reported 12 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 4 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 8 minor

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [CommunityPageSpecialController.class.php#L3](https://github.com/Wikia/app/blob/4e0bfff684a487dab5dd5d0272fa8a7f1ed35976/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L3): Class "CommunityPageSpecialController" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [ext.embeddableDiscussions.js#L10](https://github.com/Wikia/app/blob/4e0bfff684a487dab5dd5d0272fa8a7f1ed35976/extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js#L10): Function has 8 parameters which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [ext.embeddableDiscussions.js#L69](https://github.com/Wikia/app/blob/4e0bfff684a487dab5dd5d0272fa8a7f1ed35976/extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js#L69): Restrict what this loop acts on by testing each property. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AForIn)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L38](https://github.com/Wikia/app/blob/4e0bfff684a487dab5dd5d0272fa8a7f1ed35976/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L38): Define a constant instead of duplicating this literal "modules" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L47](https://github.com/Wikia/app/blob/4e0bfff684a487dab5dd5d0272fa8a7f1ed35976/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L47): Define a constant instead of duplicating this literal "CommunityPageSpecialController" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L153](https://github.com/Wikia/app/blob/4e0bfff684a487dab5dd5d0272fa8a7f1ed35976/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L153): Define a constant instead of duplicating this literal "noAdminContactText" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L153](https://github.com/Wikia/app/blob/4e0bfff684a487dab5dd5d0272fa8a7f1ed35976/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L153): Define a constant instead of duplicating this literal "communitypage-no-admins-contact" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L353](https://github.com/Wikia/app/blob/4e0bfff684a487dab5dd5d0272fa8a7f1ed35976/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L353): Define a constant instead of duplicating this literal "timeAgo" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

rybmat committed:
parenthesis

sebastianmarzjan commented on extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php:
```diff
@@ -56,6 +58,10 @@ public function index() {
 			'communityTodoListModule' => $this->getCommunityTodoListData(),
+			'discussionsModule' => !empty( $wgEnableDiscussions ) ? ( new EmbeddableDiscussionsController() )->render( [
+				'columns' => static::DISCUSSIONS_MODULE_COLUMNS,
+				'size' => static::DISCUSSIONS_MODULE_SIZE
```
👍 for extracting those and making them integers

dianafa committed:
spacing JS

WikiaTech commented on extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js:
```diff
@@ -7,7 +7,7 @@ require([
 	'embeddablediscussions.templates.mustache',
 	'EmbeddableDiscussionsSharing'
-], function ($, tracker, uiFactory, mustache, window, throbber, templates, sharing) {
+], function($, tracker, uiFactory, mustache, window, throbber, templates, sharing) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Function has 8 parameters which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)

WikiaTech commented:
SonarQube analysis reported 12 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 4 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 8 minor

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [CommunityPageSpecialController.class.php#L3](https://github.com/Wikia/app/blob/94b12bbec74197c0b7af7baee1065cbdbfb63be1/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L3): Class "CommunityPageSpecialController" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [ext.embeddableDiscussions.js#L69](https://github.com/Wikia/app/blob/94b12bbec74197c0b7af7baee1065cbdbfb63be1/extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js#L69): Restrict what this loop acts on by testing each property. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AForIn)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L38](https://github.com/Wikia/app/blob/94b12bbec74197c0b7af7baee1065cbdbfb63be1/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L38): Define a constant instead of duplicating this literal "modules" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L47](https://github.com/Wikia/app/blob/94b12bbec74197c0b7af7baee1065cbdbfb63be1/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L47): Define a constant instead of duplicating this literal "CommunityPageSpecialController" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L153](https://github.com/Wikia/app/blob/94b12bbec74197c0b7af7baee1065cbdbfb63be1/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L153): Define a constant instead of duplicating this literal "noAdminContactText" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L153](https://github.com/Wikia/app/blob/94b12bbec74197c0b7af7baee1065cbdbfb63be1/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L153): Define a constant instead of duplicating this literal "communitypage-no-admins-contact" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L353](https://github.com/Wikia/app/blob/94b12bbec74197c0b7af7baee1065cbdbfb63be1/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L353): Define a constant instead of duplicating this literal "timeAgo" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

sebastianmarzjan commented on extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php:
```diff
@@ -36,134 +44,168 @@ public static function onBeforePageDisplay() {
+			return $this->renderError( $errorMessage );
+		}
+
+		return F::app()->checkSkin( 'wikiamobile' ) ?
```
👍  - that usage of trinary operator warms my heart :)

sebastianmarzjan commented on extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js:
```diff
@@ -101,21 +101,18 @@ require([
 
 		// Inject proper class for 2 columns display
 		if (requestData.columns === 2) {
```
That one thing might be a const, though in ES5 world of Oasis I'd say we pass

sebastianmarzjan commented on extensions/wikia/EmbeddableDiscussions/styles/EmbeddableDiscussions.scss:
```diff
@@ -12,21 +14,24 @@ $upvote-icon-tiny: '/extensions/wikia/EmbeddableDiscussions/images/upvote-tiny.s
+$content-spacing-small-plus: 14px;
+$content-spacing-regular: 18px;
+$content-spacing-large: 24px;
+$content-spacing-xxlarge: 48px;
```
Can you link a ticket to tech-debt removal one? It seems we'll have those defined in two places

sebastianmarzjan commented on extensions/wikia/EmbeddableDiscussions/styles/EmbeddableDiscussions.scss:
```diff
@@ -53,48 +58,39 @@ $font-size-large: 16px;
+		display: -webkit-box;
+		font-size: $typescale-size-minus-1;
+		-webkit-line-clamp: 3;
+		overflow: hidden;
```
How is the cut-off supported on non-webkit browsers?

sebastianmarzjan commented on extensions/wikia/EmbeddableDiscussions/styles/EmbeddableDiscussions.scss:
```diff
@@ -104,50 +100,53 @@ $font-size-large: 16px;
 
 	.embeddable-discussions-post-detail {
 		border: 1px solid $card-border-color;
 		margin: 0 0 20px;
```
Could we get rid of the 20px magic number?

sebastianmarzjan commented on extensions/wikia/EmbeddableDiscussions/styles/EmbeddableDiscussions.scss:
```diff
@@ -104,50 +100,53 @@ $font-size-large: 16px;
 	}
 
 	.embeddable-discussions-post-detail-columns {
 		margin-right: 10px;
```
Could we get rid of the 10px magic number?

sebastianmarzjan commented on extensions/wikia/EmbeddableDiscussions/styles/EmbeddableDiscussions.scss:
```diff
@@ -194,15 +194,15 @@ $font-size-large: 16px;
-		font-size: $font-size-small;
+		color: $color-light-gray;
+		font-size: $typescale-size-minus-2;
 		padding-left: 5px;
```
Could we get rid of the 5px magic number?

sebastianmarzjan commented on extensions/wikia/EmbeddableDiscussions/styles/EmbeddableDiscussions.scss:
```diff
@@ -219,8 +219,8 @@ $font-size-large: 16px;
 		background: url($upvote-icon-tiny) no-repeat;
-		margin-right: 4px;
 		height: $tiny-icon-height;
+		margin-right: 4px;
```
Could we get rid of the 4px magic number?

sebastianmarzjan commented on extensions/wikia/EmbeddableDiscussions/styles/EmbeddableDiscussions.scss:
```diff
@@ -231,19 +231,19 @@ $font-size-large: 16px;
 		clear: both;
-		padding: 48px 0 75px;
 		line-height: 1.4;
+		padding: 48px 0 75px;
```
Could we get rid of the magic numbers?

sebastianmarzjan commented on extensions/wikia/EmbeddableDiscussions/styles/avatar.scss:
```diff
@@ -1,9 +1,13 @@
 .embeddable-discussions-module {
 	.avatar-container {
+		background-image: url('/extensions/wikia/GlobalNavigation/images/signin_icon.svg'); /* inline */
+		background-position: center 2px;
```
Let's leave a comment why we need to reposition the background (shape of default avatar)

rybmat commented on extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php:
```diff
@@ -36,134 +44,168 @@ public static function onBeforePageDisplay() {
-			$args['mostrecent'] !== 'true' &&
-			$args['mostrecent'] !== 'false'
+	private function checkArguments( $modelData, array $args, &$errorMessage ) {
+		// PARAM_MOSTRECENT must be bool
```
this comment is misleading, as it is defined as string

rybmat commented on extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php:
```diff
@@ -36,134 +44,168 @@ public static function onBeforePageDisplay() {
 			return false;
 		}
 
 		// size must be integer in range
```
what range?

rybmat commented on extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php:
```diff
@@ -36,134 +44,168 @@ public static function onBeforePageDisplay() {
 			}
 		}
 
 		// columns must be integer in range
```
what range?

rybmat commented on extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php:
```diff
@@ -36,134 +44,168 @@ public static function onBeforePageDisplay() {
 			}
 		}
 
 		// category must be a valid category
```
<3 such comments. What is considered to be valid category?

lukaszkonieczny committed:
Merge remote-tracking branch 'origin/dev' into WW-99

idradm committed:
introduce fandom solr search

Commit:
Add caching, remove proxy class

pgrabowski-wikia committed:
formatting and logging

pgrabowski-wikia committed:
formatting and logging

Commit:
XW-2101 | Escape message

pgrabowski-wikia committed:
Merge pull request #11429 from Wikia/PLATFORM-2429

PLATFORM-2429 adding confirmation in case a searched element does not exist

Commit:
XW-2101 | Fix unit tests for Lang::ShortenNumberDecorator

Commit:
XW-2101 | Use Lang::shortenNumberDecorator in Search

Commit:
XW-2101 | Remove shortNumForMsg

Commit:
XW-2101 | Adjust shortenNumberDecorator

macbre committed:
WikisApiController: fix default values handling

WikisApiController did not have a default value for thumbnail height and was not handling zero value for width (passed by the old mobile apps).

* fix default values handling (by using a default when zero is passed for either width or height)
* actually define a default for height

These changes will be applied to getDetails method

macbre committed:
WikisApiController: get rid of unused $flagsBlacklist

macbre committed:
DPLMain::getSubcategories - properly escape category name

macbre committed:
DPLMain: use NS_CATEGORY const instead of hardcoded value

Commit:
ADEN-3994 rename ahspace to ah

Commit:
XW-2101 | Search exact match - style stats

Commit:
XW-2101 | Escape WikiaSearch_exactWikiMatch template

fraszczakszymon committed:
ADEN-3990 Dispatch uap/not_uap events from AdDetect

Commit:
PLATFORM-410: Extra logging for deadlocks

macbre committed:
Merge pull request #11443 from Wikia/PLATFORM-2427-WikisApi-default-values-handling

WikisApiController: fix default values handling

Commit:
XW-2101 | Remove unused variable

macbre committed:
Merge pull request #11444 from Wikia/PLATFORM-2422-dpl-sql-injection-fix

	DPLMain::getSubcategories - properly escape category name

Commit:
Merge pull request #11428 from Wikia/XW-1807

XW-1807 | remove code for old GlobalFooter

Commit:
Merge pull request #11445 from Wikia/extra-logging-for-action-delete

PLATFORM-410: Extra logging for deadlocks

Commit:
XW-2101 | Coding conventions

Commit:
XW-2101 | Fix whitespaces

Commit:
XW-2101 | Tracking.js | e -> event

Commit:
Merge branch 'XW-search' into XW-search-fandom

fraszczakszymon committed:
ADEN-3990 uapContext unit tests

Commit:
ADEN-3930 move rules to WikiaRobots

Commit:
XW-2101 | Cleanup fandom search

Commit:
XW-2101 | Search urlencode query param for links

Commit:
MAIN-8064 in ArticleCommentsAjax rename isValidWriteRequest to assertValidWriteRequest

Grunny committed:
XW-1925 | Fix support for links in zh-hk and other Chinese variants

By changing the array keys to zh-hans and zh-hant we allow all of MediaWiki's
fallbacks for Chinese language codes to work for links in the DesignSystem API.

Commit:
Merge remote-tracking branch 'origin/XW-search' into XW-search-fandom

Commit:
XW-2101 | Whitespaces

Commit:
XW-2101 | Simplify condition

Commit:
XW-2107 | added input type='search' to search input for compatibility with Mercury; closing search input on keydown instead of keyup;

Commit:
XW-2101 | Cleanup WikiaSearchHelper

Commit:
XW-2101 | Wikia\Search\MediaWikiService remove a line

Commit:
XW-2101 | Wikia\Search\MediaWikiService remove a line

Commit:
Merge pull request #11450 from Wikia/XW-1925

XW-1925 | Fix support for links in zh-hk and other Chinese variants

Commit:
XW-2101 | Simplify WikiaSearchHelper

Commit:
Merge pull request #11449 from Wikia/MAIN-8064

MAIN-8064 rename isValidWriteRequest to assertValidWriteRequest

bulaszenko committed:
Merge pull request #11419 from Wikia/ADEN-3594

ADEN-3594 Vast improvements

macbre committed:
FSFile: log when sha1_file fails

Let's debug "PHP Warning: mwstore://swift-backend/X was not stored with SHA-1 metadata. in /includes/filerepo/backend/SwiftFileBackend.php on line 574" which is most likely caused by the above (a local file is missing?)

macbre committed:
SwiftFileBackend: log more context with SHA1 calculation warnings

Commit:
XW-2107 | ds upgrade

Commit:
XW-2101 | Download translations

macbre committed:
Merge pull request #11453 from Wikia/PLATFORM-1736-swift-log-missing-sha1

PLATORM-1736 - 	FSFile: log when sha1_file fails

Commit:
XW-2107 | updated ds version do 0.14.0

Commit:
Merge pull request #11430 from Wikia/XW-2107

XW-2107 | close Global Nav search input on Escape key press

mklucsarits committed:
Merge pull request #11424 from Wikia/FAN-1027

FAN-1027 Use different links in global nav for user on Fandom stories

Commit:
Merge pull request #11352 from Wikia/cp-redesign2

Community Page redesign feature branch

Commit:
XW-2119 | fix default link to Facebook

Commit:
XW-2119 | add Facebook link for Japanese

Commit:
Merge pull request #11291 from Wikia/ADEN-3930-google-bot

ADEN-3639 do not allow google bot to index this controller method

Commit:
Revert "XW-2119 | add Facebook link for Japanese"

This reverts commit a4d81f2ec6e033b8a0419cd00c2a472457523f70.

kerriamber committed:
Merge pull request #11456 from Wikia/release-444.003-MAIN-8063

MAIN-8063: Most of the features button in source editor are not present on preview

kerriamber committed:
Merge pull request #11457 from Wikia/dev-MAIN-8063

backport fix for MAIN-8063 to dev

Commit:
Merge pull request #11455 from Wikia/XW-2119

XW-2119 | fix default link to Facebook

Commit:
SEO-548 Hotfix for paragon.wiki -> paragon.wikia.com redirect

Commit:
SEO-548 Better check for upper case and removed out of bound problem

Commit:
Merge pull request #11460 from Wikia/SEO-548-paragon-wiki-redirects-hotfix

SEO-548 Hotfix for paragon.wiki -> paragon.wikia.com redirect

Commit:
SUS-939 rebranding: replace staff signature image

Commit:
use global variable

Commit:
adjust image size in CSS

Commit:
Merge pull request #11466 from Wikia/SUS-939

Sus 939

Commit:
Revert "Sus 939"

Commit:
Merge pull request #11467 from Wikia/revert-11466-SUS-939

Revert "Sus 939"

bulaszenko committed:
Merge pull request #11448 from Wikia/ADEN-3990

ADEN-3990 Refactor wikia.uap and wikia.not_uap events

Commit:
Merge pull request #11308 from Wikia/SOC-3000

SOC-3000: Allow root level paths in Oasis nav

Commit:
XW-2118 | hide fandom stories behind flag

Commit:
XW-1936 | remove 'API' link from global footer

Commit:
XW-1936 | update DS-i18n to 0.6.0

Commit:
XW-2131 | Language:shortenNumberDecorator() fix plurals

Commit:
Merge branch 'XW-search' of github.com:Wikia/app into XW-search

Commit:
XW-2101 | Whitespaces

Commit:
XW-2101 | fulltext and resultsLang added to the full results link

Commit:
XW-2101 | remove fulltext query param

Commit:
XW-2101 | build query in a prettier way

Commit:
XW-2132 | Fix search suggestions when foced search too fast

Commit:
XW-2132 | Search Suggestions polishing JS

Commit:
XW-2101 | add '?' to add the query

Commit:
Merge remote-tracking branch 'origin/XW-search' into XW-search-fandom

Commit:
Merge pull request #11451 from Wikia/XW-search-fandom

XW-2101 | Use Solr for fandom search

Commit:
Merge pull request #11474 from Wikia/XW-2132

XW-2132 | Fix search suggestions when foced search too fast

Commit:
PLATFORM-410: Skip INSERT IGNORE if category row exists

Commit:
More readable code

Commit:
RECIRC-213: Remove scroll depth test

Commit:
Merge pull request #11477 from Wikia/RECIRC-213

RECIRC-213: Remove scroll depth test

Commit:
XW-2101 | Search - Fixed passed parameters for wfMessage

Commit:
Merge pull request #11475 from Wikia/skip-inserts-when-category-row-exists

PLATFORM-410: Skip INSERT IGNORE if category row exists

Commit:
XW-2134 | show view more fandom stories link for more that 5 results

Commit:
Merge pull request #11473 from Wikia/XW-1936

XW-1936 | remove 'API' link from global footer

Commit:
XW-2135 | init search suggestions when DOM is ready

Commit:
Merge pull request #11372 from Wikia/XW-search

XW-2101 | Search Result Page improvements

Commit:
Merge pull request #11481 from Wikia/XW-2135

XW-2135 Init search suggestions when DOM is ready

Commit:
XW-2106 | enhance readability of Chinese Start a Wiki button in nav

Commit:
XW-2121 | show all 5 social network icons on all languages

Commit:
XW-2121 | add trailing coma

macbre committed:
SUS-1012: introduce CompareMessagingWiki

macbre committed:
Import ImportMessagingWiki by @TK-999

macbre committed:
ImportMessagingWiki: get all i18n files and allow filtering by a single message key

Commit:
XW-2106 | adjust ZH SAW button paddings

bulaszenko committed:
Merge pull request #11433 from Wikia/ADEN-3994

ADEN-3994 calc the page height

Commit:
Merge pull request #11485 from Wikia/XW-2121

XW-2121 | show all 5 social network icons on all languages

Commit:
XW-2106 | install design-system@0.14.2

Commit:
Merge pull request #11484 from Wikia/XW-2106

XW-2106 | enhance readability of Chinese Start a Wiki button in nav

macbre committed:
Import autocreatewiki-welcomebody from Messaging Wiki using @TK-999 tool

Commit:
XW-2140 | button :focus state fix

Commit:
XW-2140 | button :focus:disabled state fix

macbre committed:
ImportMessagingWiki: get rid of articlesToDelete

macbre committed:
ImportMessagingWiki: escape messages content

macbre committed:
CreateWikiChecks.i18n.php: escape messages

macbre committed:
Keep proper spacing

macbre committed:
ImportMessagingWiki: escape only apostophes

macbre committed:
Import autocreatewiki-welcometalk from Messaging Wiki using @TK-999 tool

macbre committed:
Import autocreatewiki-welcometalk-wall from Messaging Wiki using @TK-999 tool

macbre committed:
Import forum-policies-and-faq from Messaging Wiki using @TK-999 tool

macbre committed:
Import oasis-anon-user from Messaging Wiki using @TK-999 tool

macbre committed:
Import passwordremindertext from Messaging Wiki using @TK-999 tool

macbre committed:
prefs-help-email: fix syntax error

macbre committed:
Import phalanx-user-block-reason-exact from Messaging Wiki using @TK-999 tool

macbre committed:
Import phalanx-user-block-reason-ip from Messaging Wiki using @TK-999 tool

macbre committed:
Import phalanx-user-block-reason-similar from Messaging Wiki using @TK-999 tool

macbre committed:
Import shared_help_edit_info from Messaging Wiki using @TK-999 tool

macbre committed:
Import specialcontact-intro-blocked from Messaging Wiki using @TK-999 tool

macbre committed:
Import specialcontact-intro-content-issue-mobile from Messaging Wiki using @TK-999 tool

macbre committed:
Import specialcontact-intro-general from Messaging Wiki using @TK-999 tool

macbre committed:
Import specialcontact-mail from Messaging Wiki using @TK-999 tool

macbre committed:
Import specialcontact-mailsub from Messaging Wiki using @TK-999 tool

macbre committed:
Import specialcontact-submitcomplete from Messaging Wiki using @TK-999 tool

macbre committed:
Import specialcontact-wikiname from Messaging Wiki using @TK-999 tool

macbre committed:
Import tagline-url from Messaging Wiki using @TK-999 tool

macbre committed:
Fix syntax error

macbre committed:
Import tagline-url-interwiki from Messaging Wiki using @TK-999 tool

macbre committed:
Import welcome-message-anon from Messaging Wiki using @TK-999 tool

macbre committed:
Import elcome-message-anon-staff from Messaging Wiki using @TK-999 tool

Commit:
XW-2140 | button Oasis fixes for different states

macbre committed:
Import welcome-message-user from Messaging Wiki using @TK-999 tool

macbre committed:
Import welcome-message-user-staff from Messaging Wiki using @TK-999 tool

macbre committed:
Import welcome-message-wall-anon from Messaging Wiki using @TK-999 tool

macbre committed:
Import welcome-message-wall-anon-staff from Messaging Wiki using @TK-999 tool

macbre committed:
Import welcome-message-wall-user from Messaging Wiki using @TK-999 tool

macbre committed:
Import welcome-message-wall-user-staff from Messaging Wiki using @TK-999 tool

macbre committed:
Import wikia-visualeditor-dialog-wikiamediainsert-policy-message from Messaging Wiki using @TK-999 tool

owend committed:
Merge pull request #11439 from Wikia/SUS-1048

SUS-1048: cap date range instead of throwing exception

Commit:
XW-2140 | install Wikia/design-system#0.14.3

Commit:
Merge pull request #11489 from Wikia/XW-2140

XW-2140 | fixes for different WDS button states

jsutterfield committed:
SOC-2815 download translations

Commit:
FAN-1047 Don't show verticals in global nav for non-EN

Commit:
Merge pull request #11376 from Wikia/SOC-2815

SOC-2815 Allow Special:Forums to be disabled and Forum:Index to be archived at the same time

mklucsarits committed:
Merge pull request #11491 from Wikia/FAN-1047

FAN-1047 Don't show verticals in global nav for non-EN

macbre committed:
Remove maintenance/wikia/compareMessagingWiki.php

It was a temporary maintenance script

macbre committed:
Merge branch 'dev' into SUS-1012

Conflicts:
	extensions/wikia/Forum/Forum.i18n.php

macbre committed:
Import forum-policies-and-faq from Messaging Wiki using @TK-999 tool (there was a conflict on Forum i18n when merging dev changes)

macbre committed:
ImportMessagingWiki: handle messages customized on Messaging Wiki that do not exist in i18n files

macbre committed:
Import missing i18n files entries for autocreatewiki-welcomebody-HTML

macbre committed:
Import missing i18n files entries for passwordremindertext-HTML

bkoval committed:
Merge pull request #11305 from Wikia/SOC-emails-rebranding

Social | Rebranding Emails

hakubo committed:
XW-2110 | download new messages for DesignSystem

macbre committed:
Merge branch 'dev' into SUS-1012

Conflicts:
	extensions/wikia/Oasis/Oasis.i18n.php

Commit:
Merge pull request #11493 from Wikia/XW-2110

XW-2110 | download new messages for DesignSystem

mleczey committed:
Merge pull request #11494 from Wikia/SOC-emails-rebranding

SOC | emails rebranding

macbre committed:
Merge branch 'dev' into SUS-1012

Conflicts:
	extensions/wikia/Oasis/Oasis.i18n.php

macbre committed:
Merge pull request #11486 from Wikia/SUS-1012

	ImportMessagingWiki: get all i18n files and allow filtering by a single message key

sebastianmarzjan committed:
Merge pull request #11300 from Wikia/WW-99

WW-99 | Tracking for Create New Wiki

dianafa committed:
Merge branch 'dev' into WW-200diana

WikiaTech commented:
SonarQube analysis reported 12 issues
- ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 4 major
- ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 8 minor

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [CommunityPageSpecialController.class.php#L3](https://github.com/Wikia/app/blob/c6f4a3920bd1b370cd487843a9529db95ce28283/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L3): Class "CommunityPageSpecialController" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
2. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [ext.embeddableDiscussions.js#L69](https://github.com/Wikia/app/blob/c6f4a3920bd1b370cd487843a9529db95ce28283/extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js#L69): Restrict what this loop acts on by testing each property. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AForIn)
3. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L38](https://github.com/Wikia/app/blob/c6f4a3920bd1b370cd487843a9529db95ce28283/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L38): Define a constant instead of duplicating this literal "modules" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
4. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L47](https://github.com/Wikia/app/blob/c6f4a3920bd1b370cd487843a9529db95ce28283/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L47): Define a constant instead of duplicating this literal "CommunityPageSpecialController" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
5. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L153](https://github.com/Wikia/app/blob/c6f4a3920bd1b370cd487843a9529db95ce28283/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L153): Define a constant instead of duplicating this literal "noAdminContactText" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
6. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L153](https://github.com/Wikia/app/blob/c6f4a3920bd1b370cd487843a9529db95ce28283/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L153): Define a constant instead of duplicating this literal "communitypage-no-admins-contact" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
7. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [CommunityPageSpecialController.class.php#L353](https://github.com/Wikia/app/blob/c6f4a3920bd1b370cd487843a9529db95ce28283/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L353): Define a constant instead of duplicating this literal "timeAgo" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

dianafa commented:
new PR to this ticket: https://github.com/Wikia/app/pull/11506
